### DISCUSSION TITLE
[KML] Style ignored?

### DISCUSSION THREAD

**Discussion by Shohreh** [organicmaps/organicmaps#6293] [Q&A] [answered]
Helloj

I tried using the code below to turn a track red, but it's still displayed using the default blue color after importing it in OM.

Am I doing it wrong, or doesn't OM support that syntax?

Thank you.

```
<?xml version="1.0" encoding="utf-8"?>
<kml xmlns="http://www.opengis.net/kml/2.2">
  <Document>
    <Style id="red">
        <LineStyle>
          <color>red</color>
          <width>6</width>
        </LineStyle>
    </Style>
  
    <Placemark>
      <name>Some track</name>
      <styleUrl>#red</styleUrl>
      <LineString>
        <coordinates>2.168875,41.373311,27.25 2.168516,41.373416,28.75 2.168432,41.37344,29 2.168484,41.373493,28.5 2.168961,41.373944,25.25 2.168972,41.373954,25.25 2.16913,41.374104,24.25 2.169229,41.374197,23.75 2.16954,41.374481,22 2.169861,41.374782,18.5 2.169901,41.37482,18 2.169982,41.374878,17 2.170029,41.374961,16.25 2.169721,41.374969,18.5 2.169663,41.37497,18.75 2.168077,41.37498,28.5 2.168014,41.37498,28.25 2.167955,41.37498,28 2.16794,41.374981,28 2.167892,41.374982,27.75 2.167646,41.374981,26.5 2.167626,41.374981,26.5 2.167572,41.374981,26.25 2.167518,41.374981,26 2.165724,41.374988,22.25 2.165682,41.374988,22.5 2.165632,41.374987,22.5 2.165478,41.374986,23.25 2.165376,41.374988,23.5 2.165358,41.374989,23.5 2.165302,41.374989,23.75 2.165265,41.374989,24 2.163482,41.374998,29.75 2.163432,41.374997,29.75 2.163196,41.374997,30.25 2.163019,41.374997,31 2.162997,41.374997,31 2.162929,41.374997,31.25 2.162511,41.374996,32.75 2.161733,41.375,31 2.161705,41.375001,31 2.161632,41.375001,30.75 2.161611,41.375001,30.75 2.161517,41.375002,30.75 2.161165,41.375004,30.75 2.161121,41.375004,30.75 2.161061,41.375004,30.75 2.161047,41.375004,30.75 2.160935,41.375004,30.75 2.160238,41.375005,35.75 2.160222,41.375005,36 2.160166,41.375005,36.5 2.160106,41.375005,37 2.158907,41.375004,35.25 2.158855,41.375004,35 2.158811,41.375005,35 2.158785,41.375005,35 2.15865,41.375003,34.75 2.158493,41.375006,34.25 2.158467,41.375007,34.25 2.158412,41.375007,34 2.158378,41.375007,34 2.156682,41.375013,34 2.156631,41.375014,34 2.156576,41.375014,34 2.156549,41.375014,34 2.156183,41.37501,34 2.155816,41.375014,34 2.155799,41.375014,34 2.155729,41.375014,34 2.155698,41.375014,34 2.154543,41.375015,36.75 2.154513,41.375015,36.75 2.154461,41.375013,37.25 2.154409,41.375014,37.5 2.154144,41.375016,39 2.153982,41.375016,39.25 2.153804,41.375016,39.75 2.153752,41.375016,39.75 2.153698,41.375016,40 2.153539,41.375016,40.25 2.153439,41.375017,40.5 2.153277,41.375018,40.75 2.152291,41.37502,36.75 2.152258,41.37502,36.75 2.152199,41.375021,36.5 2.1522,41.375082,36.25 2.152201,41.375126,36.25 2.152199,41.375188,36.25 2.152199,41.375208,36.25 2.152198,41.375242,36.25 2.152197,41.375274,36.25 2.152154,41.375275,36 2.152128,41.375282,36 2.151985,41.375286,35.25 2.151794,41.375288,34.5 2.151667,41.375382,34 2.150906,41.375955,37 2.150792,41.376042,37.25 2.150655,41.376148,37 2.150629,41.376172,37 2.150558,41.376231,37 2.150527,41.376253,37 2.150471,41.376291,37 2.150386,41.376362,37 2.150236,41.376475,37.25 2.149635,41.376926,34.25 2.149521,41.377018,33.5 2.149406,41.377105,32.75 2.149363,41.377073,32.75 2.14909,41.376871,32.25 2.149018,41.37681,32.5 2.148483,41.376399,36 2.148402,41.376329,37.25 2.148386,41.376285,37.75 2.148388,41.376261,37.75 2.148357,41.376244,38.25 2.148273,41.376197,39.5 2.148119,41.376113,41.75 2.148095,41.376106,42 2.148052,41.376076,42.75 2.148028,41.376095,42.75 2.147907,41.37619,43.5 2.14789,41.376203,43.75 2.147738,41.376323,44.75 2.147209,41.376737,45.75 2.146971,41.376911,43.5 2.14687,41.376984,42.5 2.146854,41.376996,42.5 2.14683,41.377013,42.25 2.146778,41.377051,41.75 2.14663,41.377161,40.5 2.146358,41.377363,39 2.145907,41.377723,39.5 2.14575,41.377836,39.75 2.145633,41.377919,39.75 2.145602,41.377941,40 2.145586,41.377953,40 2.145452,41.378053,40.5 2.144787,41.378551,42.25 2.144621,41.378675,41.75 2.144601,41.378691,41.75 2.144559,41.378721,41.75 2.144549,41.378729,41.75 2.144457,41.378798,41.5 2.143676,41.379384,40.75 2.143571,41.379461,41.25 2.143526,41.379494,41.5 2.14348,41.37953,41.75 2.143014,41.379889,41.75 2.142761,41.380078,41 2.142714,41.380114,41 2.142659,41.380156,41 2.14243,41.38036,42 2.142241,41.380529,44 2.142138,41.380597,44.5 2.142092,41.38059,44.5 2.141582,41.380517,43.25 2.141492,41.380502,43.5 2.141407,41.380488,43.5 2.141259,41.380473,43.75 2.141108,41.380438,43.75 2.141064,41.38043,43.75 2.141026,41.380423,43.75 2.140929,41.380408,44 2.140813,41.38037,43.75 2.140766,41.380344,43 2.14074,41.380289,42.5 2.140734,41.380262,42.25 2.140743,41.380221,41.75 2.140791,41.380148,41.25 2.140907,41.380038,40 2.141063,41.379889,39.5 2.141175,41.379785,39.5 2.141261,41.379706,39.25</coordinates>
      </LineString>
    </Placemark>
  </Document>
</kml>
```

**Comment by biodranik** [comment]
The color in KML should be specified in the hexadecimal notation, not in plain English.

**Comment by Shohreh** [comment]
Thanks for the idea, but it's still displayed as a blue line instead of [red](https://www.computerhope.com/htmcolor.htm):

```
<?xml version="1.0" encoding="utf-8"?>
<kml xmlns="http://www.opengis.net/kml/2.2">
  <Document>
    <name>test.style.kml</name>
    <Style id="red">
        <LineStyle>
          <!-- color>red</color -->
          <!-- color>#FF0000</color -->
          <color>FF0000</color>
          <width>6</width>
        </LineStyle>
    </Style>
  
    <Placemark>
      <name>blah</name>
      <styleUrl>#red</styleUrl>
      <LineString>
        <coordinates>2.168875,41.373311,27.25 2.168516,41.373416,28.75 2.168432,41.37344,29 2.168484,41.373493,28.5 2.168961,41.373944,25.25 2.168972,41.373954,25.25 2.16913,41.374104,24.25 2.169229,41.374197,23.75 2.16954,41.374481,22 2.141175,41.379785,39.5 2.141261,41.379706,39.25</coordinates>
      </LineString>
    </Placemark>
  </Document>
</kml>
```

**Reply by biodranik** [reply]
KML colors nave 4 bytes, not 3: https://developers.google.com/kml/documentation/kmlreference#colorstyle

**Comment by Shohreh** [comment, marked as answer]
Thanks. [Still no go](https://gis.stackexchange.com/questions/444099/kml-polystyle-colour-not-showing):

    <Style id="red">
        <LineStyle>
          
          
          
           
          <color>FF00007F</color> 
          <width>6</width>
        </LineStyle>
    </Style>

Is it really AABBGGRR?

--
Edit: Read too fast. In KML, it's BGR, not the usual RGB. This does display the linestring in red:

`<color>7F0000FF</color>`

Thanks!
